shared/src/components/tabulator/NullableInputEditor.vue

```diff
@@ -10,27 +24,47 @@
       @change.prevent="submit"
       @keydown="keydown"
     >
-    <i
-      class="material-icons clear"
-      @mousedown.prevent.stop="clear"
-      title="Nullify Value"
-    >cancel</i>
+    <div class="icon-wrapper">
+      <i
+        class="material-icons special-type"
+        @mousedown.prevent.stop="toggleTypeEditor"
+        :title="this.typeEditorTitle"
+      >{{ this.typeEditorIcon }}</i>
+      <i
+        class="material-icons clear"
+        @mousedown.prevent.stop="clear"
+        title="Nullify Value"
+      >cancel</i>
+    </div>
   </div>
 </template>
 <script lang="ts">
 import _ from 'lodash'
 import Vue from 'vue'
+import DatePicker from 'vue2-datepicker'
 import helpers from '@shared/lib/tabulator'
 export default Vue.extend({
   props: ['cell', 'params'],
+  components: { DatePicker },
   data() {
     return {
       value: null,
       rendered: false,
-      everEdited: false
+      everEdited: false,
+      typeEditorActive: false
     }
   },
   computed: {
+    typeEditorTitle() {
+      if (this.typeEditorActive) return 'Use standard input'
+      if (helpers.isDateTime(this.params.dataType)) return 'Open date/time picker' 
+      return ''
+    },
+    typeEditorIcon() {
+      if (this.typeEditorActive) return 'edit'
```

Comment:
Yep, then all this stuff can move out of the NullableInputEditor and into a DatePickerEditor

Reply:
Yeah that was all part of the plan for a later PR. Was taking this approach as a get this out there and see how people like it first. I'll make the changes